client/my-sites/checkout/cart/cart-coupon.jsx

```diff
@@ -148,6 +148,19 @@ export class CartCoupon extends React.Component {
 		applyCoupon( this.state.couponInputValue );
 	};
 
+	removeCoupon = event => {
+		event.preventDefault();
+		if ( this.isSubmitting ) {
+			return;
+		}
+
+		analytics.tracks.recordEvent( 'calypso_checkout_coupon_submit', {
+			coupon_code: '',
+		} );
```

Comment:
we really don't want to issue tracking events directly. please dispatch tracking actions via Redux supplied in `connect()`

https://wpcalypso.wordpress.com/devdocs/client/state/analytics/README.md?term=analytics

Reply:
Done, thanks!